src/App.java

```diff
@@ -1,5 +1,182 @@
+        } else if (answer.equals("guess definition")) {
+            return 2;
+        } else if (answer.equals("mix")) {
+            return 3;
+        } else {
+            return -1;
+        }
+    }
+
+    /**
+     * Depending on the mode selected, this method will present to the user: definitions, words, or a mix of both.
+     */
+    public void doFlashcards() {
+        myFlashcard.shuffleDeck();
+        int mode = chooseMode();
+        while (mode == -1) { //catches the error for if the user selects a mode which doesn't exist
+            printer.output("Please select a valid mode");
+            mode = chooseMode();
+        }  
+        if (mode == 1){
+            for (int i = 0; i < myFlashcard.getNumDefinitions(); i++) { // presents the definition THEN the corresponding word
+                printer.output(myFlashcard.getDefinition(i));
+                printer.output("Please press ENTER to reveal the word");
+                printer.input();
+                printer.output(myFlashcard.getWord(i));
+                printer.output("");
+            }
+        } else if (mode == 2) {
+            for (int i = 0; i < myFlashcard.getNumWords(); i++) { // presents the word THEN the corresponding defintion
+                printer.output(myFlashcard.getWord(i));
+                printer.output("Please press ENTER to reveal the definition");
+                printer.input();
+                printer.output(myFlashcard.getDefinition(i));
+                printer.output("");
+            }
+        } else if (mode == 3){
+            for (int i = 0; i < myFlashcard.getNumWords(); i++) { // randomly shows the word or definition
+                int random = (int)(Math.random()*2);
+                if (random == 0){
+                    printer.output(myFlashcard.getWord(i));
+                    printer.output("Please press ENTER to reveal the definition");
+                    printer.input();
+                    printer.output(myFlashcard.getDefinition(i));
+                    printer.output("");
+                } else if (random == 1){
+                    printer.output(myFlashcard.getDefinition(i));
+                    printer.output("Please press ENTER to reveal the word");
+                    printer.input();
+                    printer.output(myFlashcard.getWord(i));
+                    printer.output("");
+                }
+            }
+        }
+
+    }
+
+    /**
+     * Allows the user to restart the program and play again.
+     * @return boolean stating if the user wants to restart the program.
+     */
+    public boolean replay(){
+        boolean valid = false;
+        while (!valid){
+            printer.output("");
+            printer.output("Would you like to play again?");
+            printer.output("(Please type \"Yes\" or \"No\")");
+            String replay = printer.input();
+            if (replay.toLowerCase().equals("yes")){
+                valid = true;
+                return true;

```

Comment:
since you are returning early in two of the three cases, I don't know if the `valid` flag is doing anything useful here. It could probably be replaced with a `while (true)`. Sometimes I recommend avoid `while(true)` but in this case the `while (!valid)` isn't actually doing anything different than a `while (true)`